time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateArbitrary.java

```diff
@@ -120,15 +121,28 @@ private LocalDate effectiveMinDate() {
 			effective = effective.withDayOfMonth(1).withMonth(earliestMonth);
 		}
 		if (dayOfMonthBetween.getMin() != null && dayOfMonthBetween.getMin() > effective.getDayOfMonth()) {
-			try {
+			if (isValidDate(effective.getYear(), effective.getMonth(), dayOfMonthBetween.getMin())) {
```

Comment:
I know I suggested to get rid of try-catch where appropriate. Here however you try (maybe successfully maybe not) to mimick what effective.withDayOfMonth() does in the library. I don't think that's well invested code.

My suggestion was to get rid of it OR DOCUMENT why an exception can occur here.